### PR TITLE
[FSK-25]: Integrate Prisma

### PR DESCRIPTION
**Pull request by mhmda-83**
fix configuration bug which prevents application to listen on port
add Prisma service and module (global)
add Prisma seed script to create an admin

### CODE REVIEW

File: src/config/configuration.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export interface AppConfig {
   port?: number;
 }
 
-export default registerAs('app', () => {
+export default registerAs('', () => {
```
Comment (ASafaeirad) [inline src/config/configuration.ts]: Is an empty string okay here?

File: src/prisma/prisma.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable fp/no-let */
```
Comment (ASafaeirad) [inline src/prisma/prisma.service.spec.ts:1]: I'll disable this rule for tests in our community rules.
In the next version we can remove this comment

File: src/config/configuration.ts (hunk 1 above)
Comment (MohammadForoutan) [reply src/config/configuration.ts]: `app` is for namespace and managing multiple configurations.  
if we have only one configuration and namespace we can remove `registerAs`  function too.

Comment (MohammadForoutan) [reply src/config/configuration.ts:16]: `app` is for namespace and managing multiple configurations.  
if we have only one configuration and namespace we can remove `registerAs`  function too.

Comment (mhmda-83) [reply src/config/configuration.ts]: > Is an empty string okay here?

yes it's ok

**Review by ASafaeirad — CHANGES_REQUESTED** [review]
Please remove this merge commit from your changes.

File: src/app.module.ts [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
       envFilePath: [path.join(process.cwd(), `.${getNodeEnv()}.env`)],
-      load: [configuration],
+      validationSchema: Joi.object({
+        NODE_ENV: Joi.string()
+          .valid('production')
+          .valid('development')
+          .valid('test')
+          .required(),
+        PORT: Joi.number().required(),
+        DATABASE_URL: Joi.string()
```
Comment (ASafaeirad) [inline src/app.module.ts]: We should disable the linter warnings in exceptional cases.

File: src/app.module.ts [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
 
-import configuration from './config/configuration';
+import { PrismaModule } from './prisma/prisma.module';
 
 @Module({
   imports: [
     ConfigModule.forRoot({
       cache: false,
+      isGlobal: true,
       envFilePath: [path.join(process.cwd(), `.${getNodeEnv()}.env`)],
```
Comment (ASafaeirad) [inline src/app.module.ts]: I'm afraid I have to disagree with this approach.
Keeping the environment outside of the codebase is a better practice.

**Review by ASafaeirad — CHANGES_REQUESTED** [review]

File: src/app.module.ts (hunk 3 above)
Comment (mhmda-83) [reply src/app.module.ts]: for each line or for the entire file?

File: src/app.module.ts (hunk 4 above)
Comment (mhmda-83) [reply src/app.module.ts]: what's your suggestion?

Comment (ASafaeirad) [reply src/app.module.ts]: Let's discuss it in a meeting

**Review by ASafaeirad — APPROVED** [review]

**Pull request merged**